transformations/quora_trained_t5_for_qa/transformation.py

```diff
@@ -0,0 +1,55 @@
+import torch
+from transformers import T5ForConditionalGeneration, T5Tokenizer
+
+from interfaces.QuestionAnswerOperation import QuestionAnswerOperation
+from tasks.TaskTypes import TaskType
+import random
+
+"""The T5 model and code has been taken from https://huggingface.co/ramsrigouthamg/t5_paraphraser. The test cases 
+were generated by the commented script in redundant_context_for_qa. 
+
+"""
+
+
+class QuoraT5QaPairGenerator(QuestionAnswerOperation):
+    tasks = [
+        TaskType.QUESTION_ANSWERING,
+        TaskType.QUESTION_GENERATION
+    ]
+    languages = ["en"]
+    heavy = True
+
+    def __init__(self, seed=0, model_name="ramsrigouthamg/t5_paraphraser", max_len=256):
+        super().__init__(seed)
+        self.model = T5ForConditionalGeneration.from_pretrained(model_name)
+        self.tokenizer = T5Tokenizer.from_pretrained(model_name)
+        self.max_len = max_len
+        self.num_return_sequences = 3
+        torch.manual_seed(self.seed)
+        random.seed(self.seed)
+        if torch.cuda.is_available():
+            torch.cuda.manual_seed_all(self.seed)
+
+    def generate(self, context: str, question: str, answers: [str]):
+
+        text = "paraphrase: " + question + "</s>"
+        encoding = self.tokenizer.encode_plus(text, padding="longest", return_tensors="pt")
+        input_ids, attention_masks = encoding["input_ids"], encoding["attention_mask"]
+        beam_outputs = self.model.generate(
```

Comment:
It could be a helpful addition to add a filtering step of some sort for cases where the outputs are of insufficient quality.